spec/factories/middleware_servers.rb

```diff
@@ -9,4 +9,16 @@
           :class   => 'ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServer',
           :parent  => :middleware_server do
   end
+
+  factory :hawkular_middleware_server_wildfly,
+          :aliases => ['app/models/manageiq/providers/hawkular/middleware_manager/middleware_server_wildfly'],
+          :class   => 'ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareServerWildfly',
+          :parent  => :middleware_server do
+  end
+
+  factory :hawkular_middleware_server_eap,
```

Comment:
If I am not wrong, you can move those three inherited factories within the `middleware_server` factory and then remove the `parent => :middleware_server` attribute from each child factory definition.

Here a bit of more context:
https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#inheritance

Do you think it worth to change it?

Reply:
@xeviknal I think the way it is done here in the PR is the same as the rest of miq does, so imo it is not worth to change it for consistency reasons. but I'd like to hear what @agrare thinks :)

Reply:
@abonas is right we (rightly or wrongly) follow this pattern, e.g. see the orchestration_stack factory https://github.com/ManageIQ/manageiq/blob/master/spec/factories/orchestration_stack.rb
:+1: on this for consistency